docs/adr/002-application-architecture-for-configurability.md

```diff
@@ -0,0 +1,117 @@
+
+#### 1. Move Routing from the core ui library to the instance level
+Routing and what pages exist should be determined at the instance level. Currently, routing is handled at the core UI library level so supporting additional/removal of pages requires override logic. Moving this to the instance level allows developers/stakeholders to independently manage their routes and pages.
+
+#### 2. Feature components in the core ui library composed at the instance level
+Currently, the core UI library handles the composition of each page because of its control over routing. However, pages and what they render should shift to the instance level so developers/stakeholders can control what is rendered within each page view. The core UI library should expose feature components (containers) that deliver a core feature in the VEDA dashboard such as the A&E feature. 
+
+#### 3. Modularize and create smaller reusable components
+The core UI library should also expose smaller reusable components like date pickers, form elements, analysis tools, etc… (“presentational/dumb” components). This way developers/stakeholders for example can compose a page view with the A&E feature from the core UI library and consume other reusable components to construct their page view. This also now allows developers to build their own custom components on the instance side and directly incorporate them. 
+
+#### 4. Create a Data layer that manages all data fetching
+The proposed data layer is designed to handle all data fetching, including STAC calls, and MDX parsing. Think of it as a versatile data utilities library. Introducing this layer would enable components in the core UI library to remain data-agnostic. In the event of scaling to additional or different data sources, expansion can be easily accomplished by integrating them into this centralized layer. Smart components (larger and stateful) would then efficiently consume this data layer.
+> **Integrating with [stac-react](https://github.com/developmentseed/stac-react/tree/main)**: Because these React hooks manage data fetching to the STAC API using the Context API provider pattern, this could just be used directly as it is already its own data layer. We would have to decide where we wrap the Provider though. Ideally, at the instance level, we would wrap the provider at the highest level in the tree either around the router or specific view containers and then consume these hooks down the hierarchy on the instance side. The components in the core ui library would be prop driven so this way they can be truly data agnostic and accept whatever data passed in.
+
+#### 5. Introduce a new Design System
```

Comment:
If I remember correctly, I thought we agreed that Design System should be addressed as a separate matter from this refactor, as it isn't essential for the structural modification. Did I miss anything?

Reply:
Yeah we can probably remove this and start a separate ADR for the Design system 👍🏼

Reply:
Yes, we went a bit back and forth between keeping it in one and moving it out. Makes sense to move it out, though.

I created a ticket for that here: https://github.com/NASA-IMPACT/veda-ui/issues/889 and can work on it tomorrow - including responding to the change requests here, so we can get both ADRs merged asap.